File: Assignment2/program2.c
////////////////////////////////////////////////////////////////
//
//  File Name   : program2.c
//  Description : Accepts one number and prints that total number of '*' on screen 
//  Author      : Akhilesh.P.Sonavane
//  Date        : 10/05/2025
//
////////////////////////////////////////////////////////////////

#include<stdio.h>

////////////////////////////////////////////////////////////////
// 
//  Function Name : Display
//  Description   : Takes a Number and prints '*' total of that Integer
//  Input         : Integer
//  Output        : Character
//
////////////////////////////////////////////////////////////////

void Display(int iNo)
{
    int iCnt = 0;

    if(iNo <= 0)
    {
        printf("Invalid Input\n");
    }
    else
    {
    while (iNo > iCnt)
    {
        printf("*\n");
        iNo--;
    }
}
}

int main()
{
    int iValue = 0;

    printf("Enter a Number\n");
    scanf("%d",&iValue);

    Display(iValue);

    return 0;
}

File: Assignment27/program2.java
////////////////////////////////////////////////////////////////
//
//  File Name   : program2.java
//  Description : Accept a String and a Integer and returns Frequency of String
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 29/05/2025
//
////////////////////////////////////////////////////////////////

import java.util.Scanner;

////////////////////////////////////////////////////////////////
// 
//  Class Name    : Display
//  Function Name : StrNCpyX
//  Description   : Inputs String and a Integer and returns Frequency of String
//  Input         : String
//  Output        : String
//
////////////////////////////////////////////////////////////////

class Display
{
    public void StrNCpyX(char[] chVal1, char[] chVal2, int iSize, int iNo)
    {
        int iCnt = 0;

        if((chVal1 == null) || (iNo < 0))
        {
            System.out.println("Invalid Input");
            return;
        }

        while((iCnt < iNo) && (iCnt < iSize)) 
        {
            chVal2[iCnt] = chVal1[iCnt];
            iCnt++;
        }

        System.out.println("From Character "+chVal2[0]+" For Frequency "+iNo+" String is : ");

        for(iCnt = 0; iCnt < iSize; iCnt++)
        {
            System.out.print(chVal2[iCnt]);
        }
    }
}

public class program2
{
    public static void main(String args[])
    {
        int iCnt = 0;
        int iLength = 0;
        char[] chArr1 = null;
        char[] chArr2 = null;
        String sValue = null;
        int iFreq = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter String : ");
        sValue = sobj.nextLine();

        System.out.println("Enter Frequency : ");
        iFreq = sobj.nextInt();
        
        iLength = sValue.length();

        chArr1 = new char[iLength];
        chArr2 = new char[iLength];

        for(iCnt = 0; iCnt < iLength; iCnt++)
        {
            chArr1[iCnt] = sValue.charAt(iCnt);
        }

        Display dobj = new Display();

        dobj.StrNCpyX(chArr1,chArr2,iLength,iFreq);

        sobj.close();
    }
}